trpcultivate_phenoshare/src/Plugin/TripalImporter/TripalCultivatePhenoShareImporter.php

```diff
@@ -743,6 +914,721 @@ public function formValidate($form, &$form_state) {
+        $raw_row_failed = TRUE;
+        $messages[$validator_name]['status'] = 'fail';
+        $messages[$validator_name]['details'] = $this->processValidDelimitedFileFailures($failures[$validator_name]);
+      }
+      else {
+        $messages[$validator_name]['status'] = 'pass';
+      }
+    }
+
+    // ValidHeaders.
+    $validator_name = 'valid_header';
+    if (array_key_exists($validator_name, $failures)) {
+      if (!empty($failures[$validator_name])) {
+        $messages[$validator_name]['status'] = 'fail';
+        $messages[$validator_name]['details'] = $this->processValidHeadersFailures($failures[$validator_name]);
+      }
+      else {
+        $messages[$validator_name]['status'] = 'pass';
+      }
+    }
+
+    // EmptyCell.
+    $validator_name = 'empty_cell';
+    if (array_key_exists($validator_name, $failures)) {
+      if (!empty($failures[$validator_name])) {
+        $messages[$validator_name]['status'] = 'fail';
+        $messages[$validator_name]['details'] = $this->processEmptyCellFailures($failures[$validator_name]);
+      }
+      // Only pass if raw row validation didn't fail.
+      elseif (!$raw_row_failed) {
+        $messages[$validator_name]['status'] = 'pass';
+      }
+      // Otherwise, leave status as 'todo' since 1+ raw rows failed.
+    }
+
+    return $messages;
+  }
+
+  /**
+   * Process failed validation from ProjectGenusMatch into a render array.
+   *
+   * @param array $validation_result
+   *   An associative array that was returned by the ProjectGenusMatch validator
+   *   in the event of failed validation. It contains the following keys:
+   *   - 'case': a developer-focused string describing the case checked.
+   *   - 'valid': FALSE to indicate that validation failed.
+   *   - 'failedItems': an array of items that failed with the following keys.
+   *     - 'project_provided': The name of the project provided.
+   *     - 'genus_provided': The name of the genus provided.
+   *
+   * @return array
+   *   A render array of type unordered list which is used to display feedback
+   *   to the user about the case that failed and the failed items from the
+   *   input file. Each item in the list contains the project that was entered
+   *   in the form which failed validation.
+   *
+   * @throws \Exception
+   *   - If the validation_result parameter was not formatted properly.
+   *   - If the case string returned by the validator implied validation passed.
+   *   - If the case string returned by the validator is not recognized.
+   */
+  public function processProjectGenusMatchFailures(array $validation_result) {
+
+    // Check the format of the validation_result parameter.
+    $this->checkValidationStatusArray($validation_result, 'ProjectGenusMatch');
+
+    // Check for one of the expected cases.
+    if ($validation_result['case'] == 'Project does not exist') {
+      $message = 'The project provided does not exist. Please contact your administrator to have this added.';
```

Comment:
Messages being displayed to the user should not mention project, but Research Experiment instead.
```suggestion
      $message = 'The selected Research Experiment does not exist. Please contact your administrator to have this added.';
```